### PR TITLE
handle response 0x8012

### PR DESCRIPTION
**Pull request by MichaelBitard**
I assume this appeared since I updated my zigate 1 from `3.1d` to `3.21` this morning.

I saw a strange closed PR here: https://github.com/zigpy/zigpy-zigate/pull/107 and the underlying issue is this one I guess: https://github.com/zigpy/zigpy-zigate/issues/104

Is there a way to easily test what I did in home assistant? I could copy paste the api.py and application.py into the home assistant folder but it seems a little by messy right?

### CODE REVIEW

**Comment by MichaelBitard** [comment]
I know I should handle this message in the application, but I don't know what it's for, I'm looking into it right now

File: zigpy_zigate/zigbee/application.py [hunk 1]
```diff
@@ -140,7 +140,7 @@ def zigate_callback_handler(self, msg, response, lqi):
             self.handle_message(device, response[1],
                                 response[2],
                                 response[3], response[4], response[-1])
-        elif msg == 0x8011:  # ACK Data
+        elif msg == 0x8011 or msg == 0x8012:  # ACK Data
```
Comment (MichaelBitard) [inline zigpy_zigate/zigbee/application.py:143]: Would you rather have a distinct else for the 0x8012?

**Comment by max5962** [comment]
> I assume this appeared since I updated my zigate 1 from `3.1d` to `3.21` this morning.
> 
> I saw a strange closed PR here: #107 and the underlying issue is this one I guess: #104
> 
> Is there a way to easily test what I did in home assistant? I could copy paste the api.py and application.py into the home assistant folder but it seems a little by messy right?

The PR is closed because they fork the project and cherry pick (when tested in their projects) the commit into zigpy-zigate

**Comment by max5962** [comment]
https://github.com/zigbeefordomoticz/zigpy-zigate/commits/dev

**Comment by max5962** [comment]
https://github.com/zigpy/zigpy-zigate/issues/104
They looking for volunteers for testing they fork in ZHA environment.

**Comment by MichaelBitard** [comment]
I'm not really fond of using a fork, when there is no official communication from the maintainers of this repository to do so.

Is this repository unmaintained? @doudz 

If so, I can make a pull request to change the README of this repository to explain that we should move to the https://github.com/zigbeefordomoticz/zigpy-zigate fork?

**Comment by doudz** [comment]
I just have no time to work on this repo but PR are welcome :)

File: zigpy_zigate/zigbee/application.py [hunk 2]
```diff
@@ -153,6 +155,8 @@ def zigate_callback_handler(self, msg, response, lqi):
         elif msg == 0x8702:  # APS Data confirm Fail
             LOGGER.debug('APS Data confirm Fail %s %s', response[4], response[0])
             self._handle_frame_failure(response[4], response[0])
+        elif msg == 0x9999:  # ZCL event
+            LOGGER.debug('ZCL application started %s', response[0])
```
Comment (doudz) [inline zigpy_zigate/zigbee/application.py:159]: This debug message is not really understandable in my opinion, maybe something like "Extended error code %s" is better ?

**Review by doudz — CHANGES_REQUESTED** [review]

File: zigpy_zigate/zigbee/application.py (hunk 2 above)
Comment (MichaelBitard) [reply zigpy_zigate/zigbee/application.py]: I did not even knew it was an error, I thought it was a normal application response

**Comment by MichaelBitard** [comment]
> I just have no time to work on this repo but PR are welcome :)

Open source in a nutshell! :)

File: zigpy_zigate/zigbee/application.py (hunk 2 above)
Comment (pipiche38) [reply zigpy_zigate/zigbee/application.py]: Logging an Error as a Debug looks strange

Comment (MichaelBitard) [reply zigpy_zigate/zigbee/application.py]: Do you have somewhere a documentation that could help me understand this 0x9999 event and what it means?

I have a bunch of them at each restart of home-assistant, I thought it was normal.

Is it ok if I raise this to a warning level @pipiche38?

**Comment by pipiche38** [comment]
@MichaelBitard , for the documentation I would refer to https://www.nxp.com/docs/en/user-guide/JN-UG-3113.pdf
and if you go to section 10.2.5 you'll get the list of Extended Errors

In short these are Errors code trigger by the stack itself, which comes to provide more details on an error happening.
This could be related to a plugin command, but it could also be generated by the stack it self, when handling some incoming commands/messages from the zigbee network itself.

For instance Extended error code 0x80, 0x81, 0x82, 0x83 are showing some inconsistency or wrong sizing of the PDM.

I do agree with you that warning would probably be good.

**Comment by MichaelBitard** [comment]
Thanks @pipiche38 that makes sense!

**Review by doudz — APPROVED** [review]

**Comment by doudz** [comment]
Looks good, thanks !

**Pull request merged**